docker/scripts/image_create.sh

```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/env bash
+
+###############################################################################
+# Copyright 2017 The Apollo Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+###############################################################################
+
+if [ $# != 1 ];then
+  echo 'Usage:
+  ./image_create.sh [OPTION]'
+  echo 'Options:
+  dev: create dev image
+  run: create release base image
+  '
+  exit 1
+fi
+
+TYPE=$1
+echo $TYPE
+TIME=$(date  +%Y%m%d_%H%M)
+if [ -z "${DOCKER_REPO}" ]; then
+    DOCKER_REPO=apolloauto/apollo
+fi
+
+
+APOLLO_ROOT="$( cd "$( dirname "${BASH_SOURCE[0]}" )/../.." && pwd )"
+ARCH=$(uname -m)
+TAG="${TYPE}-${ARCH}-${TIME}"
+
+# Build image from APOLLO_ROOT, while use the specified Dockerfile.
+docker build -t "${DOCKER_REPO}:tmp_tag" \
```

Comment:
tmp_tag?

Reply:
It is a temporary tag.
Finally it will flat everything to one layer and delete the temporary tag (See line 51 to 60)